### PR TITLE
 Use InformationVersion and assembly name for telemetry 

### PR DESCRIPTION
**Pull request by pakrym**
### Breaking change

Don't use AzureSdkClientLibrary anymore 

Before:

```
[assembly:AzureSdkClientLibrary("identity")]
```

After:

```diff
-[assembly:AzureSdkClientLibrary("identity")]
```

### CODE REVIEW

File: sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs [hunk 1]
```diff
@@ -48,15 +48,25 @@ public static HttpPipeline Build(ClientOptions options, bool bufferResponse = tr
...
+            const string PackagePrefix = "Azure.";
+
             Assembly clientAssembly = options.GetType().Assembly;
-            AzureSdkClientLibraryAttribute componentAttribute = clientAssembly.GetCustomAttribute<AzureSdkClientLibraryAttribute>();
-            if (componentAttribute == null)
+
+            AssemblyInformationalVersionAttribute versionAttribute = clientAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
+            if (versionAttribute == null)
+            {
+                throw new InvalidOperationException("Client SDK assemblies are required to have InformationalVersionAttribute");
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs]: Can we make this error message a little more helpful?  The old one included the client assembly name, but I'd love it to also say something along the lines of

`$"{nameof(AssemblyInformationalVersionAttribute)} is required on client SDK assembly '{clientAssembly.FullName}' (inferred from the use of options type '{options.GetType().FullName}')."`

as a hint to partner teams adding new client libraries.  I tried using the default `ClientOptions` type when I was first building a library and had to jump down the telemetry rabbit hole in the debugger.  😄

File: sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs [hunk 2]
```diff
@@ -3,13 +3,11 @@
...
 using System.Runtime.CompilerServices;
-using Azure.Core;
 
 [assembly: InternalsVisibleTo("Azure.Storage.Blobs.Tests, PublicKey=" +
     "0024000004800000940000000602000000240000525341310004000001000100d15ddcb2968829" +
     "5338af4b7686603fe614abd555e09efba8fb88ee09e1f7b1ccaeed2e8f823fa9eef3fdd60217fc" +
     "012ea67d2479751a0b8c087a4185541b851bd8b16f8d91b840e51b1cb0ba6fe647997e57429265" +
     "e85ef62d565db50a69ae1647d54d7bd855e4db3d8a91510e5bcbd0edfbbecaa20a7bd9ae74593d" +
     "aa7b11b4")]
-[assembly: AzureSdkClientLibrary("Storage.Blobs")]
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs]: Does the build automagically add the `AssemblyInformationalVersionAttribute`?  Or should we be adding it here as part of this change?

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs (hunk 2 above)
Comment (pakrym) [reply sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs]: Yep, it does, it's set to package version automatically.

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/AssemblyInfo.cs]: Awesome - having one less thing to touch is always great.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj [hunk 3]
```diff
@@ -23,4 +23,8 @@
     <PackageReference Include="System.Text.Json" />
     <PackageReference Include="System.Threading.Tasks.Extensions" />
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (maririos) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: Now that AssemblyInfo.cs was deleted, we don't need this in the csproj.
Whenever we write tests for this API we will need both again though...

**Review by maririos — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj (hunk 3 above)
Comment (pakrym) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: Ug, missed it. VS adds this automatically :(

**Pull request merged**